pkg/catalog/catalog.go

```diff
@@ -154,20 +170,25 @@ func New(ctx context.Context, cfg Config) (*Catalog, error) {
 		sstableManager,
 	)
 	if err != nil {
+		cancelFn()
```

Comment:
true, but the cancelFn must be called - it's a leak otherwise: from https://golang.org/pkg/context/#WithCancel -

> Canceling this context releases resources associated with it, so code should call cancel as soon as the operations running in this Context complete.